zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1191,8 +1199,15 @@ private void runAllParagraphs(NotebookSocket conn,
             new TypeToken<List<Map<String, Object>>>() {
             }.getType());
 
-    getNotebookService().runAllParagraphs(noteId, paragraphs, getServiceContext(fromMessage),
-        new WebSocketServiceCallback<Paragraph>(conn));
+    Note note = notebook().getNote(noteId);
+
+    try {
+      note.setRunning(true);
```

Comment:
Could you move this operation into `NotebookService` ?

Reply:
done